tests/test_legacy.py

```diff
@@ -283,3 +141,500 @@ def generate_common_dicom_dataset_series(self, slice_count: int,
+#             tmp_dataset.SeriesNumber = 1
+#             tmp_dataset.ImageOrientationPatient =\
+#                   [1.000000, 0.000000, 0.000000,
+#                   0.000000, 1.000000, 0.000000]
+#             tmp_dataset.ImagePositionPatient = [0.0, 0.0,
+#                                                 tmp_dataset.SliceLocation]
+#             tmp_dataset.ImageType = ['ORIGINAL', 'PRIMARY', 'AXIAL']
+#             tmp_dataset.PixelSpacing = [1, 1]
+#             tmp_dataset.PatientName = 'John Doe'
+#             tmp_dataset.FrameOfReferenceUID = frame_of_ref_uid
+#             tmp_dataset.SOPClassUID = sop_classes[system][1]
+#             tmp_dataset.SOPInstanceUID = generate_uid()
+#             tmp_dataset.SeriesInstanceUID = series_uid
+#             tmp_dataset.StudyInstanceUID = study_uid
+#             tmp_dataset.BitsAllocated = bytes_per_voxel * 8
+#             tmp_dataset.BitsStored = bytes_per_voxel * 8
+#             tmp_dataset.HighBit = (bytes_per_voxel * 8 - 1)
+#             tmp_dataset.PixelRepresentation = 1
+#             tmp_dataset.Columns = cols
+#             tmp_dataset.Rows = rows
+#             tmp_dataset.SamplesPerPixel = 1
+#             tmp_dataset.AccessionNumber = '2'
+#             tmp_dataset.AcquisitionDate = date_
+#             tmp_dataset.AcquisitionTime = datetime.now().time()
+#             tmp_dataset.AdditionalPatientHistory = 'UTERINE CA PRE-OP EVAL'
+#             tmp_dataset.ContentDate = date_
+#             tmp_dataset.ContentTime = datetime.now().time()
+#             tmp_dataset.Manufacturer = 'Mnufacturer'
+#             tmp_dataset.ManufacturerModelName = 'Model'
+#             tmp_dataset.Modality = sop_classes[system][0]
+#             tmp_dataset.PatientAge = '064Y'
+#             tmp_dataset.PatientBirthDate = date_ - age
+#             tmp_dataset.PatientID = 'ID0001'
+#             tmp_dataset.PatientIdentityRemoved = 'YES'
+#             tmp_dataset.PatientPosition = 'FFS'
+#             tmp_dataset.PatientSex = 'F'
+#             tmp_dataset.PhotometricInterpretation = 'MONOCHROME2'
+#             tmp_dataset.PixelData = pixel_array
+#             tmp_dataset.PositionReferenceIndicator = 'XY'
+#             tmp_dataset.ProtocolName = 'some protocole'
+#             tmp_dataset.ReferringPhysicianName = ''
+#             tmp_dataset.SeriesDate = date_
+#             tmp_dataset.SeriesDescription = 'test series '
+#             tmp_dataset.SeriesTime = time_
+#             tmp_dataset.SoftwareVersions = '01'
+#             tmp_dataset.SpecificCharacterSet = 'ISO_IR 100'
+#             tmp_dataset.StudyDate = date_
+#             tmp_dataset.StudyDescription = 'test study'
+#             tmp_dataset.StudyID = ''
+#             if (system == Modality.CT):
+#                 tmp_dataset.RescaleIntercept = 0
+#                 tmp_dataset.RescaleSlope = 1
+#             tmp_dataset.StudyTime = time_
+#             output_dataset.append(tmp_dataset)
+#         return output_dataset
```

Comment:
Dicom generation part of the tests now is a separate class. I also removed the mixed SeriesInstanceUID test because based on David Clunie's opinion multi-frame images can span through multiple series.  We also don't consider different transfer syntax as a distinguishing attribute, so I removed the mixed transfer syntax test too. @fedorov, please check out my comment to discuss this issue further.

Reply:
I agree SeriesInstanceUID should not be used as a distinguishing attribute. One most common example is dynamic contrast imaging, which some vendors encode as a separate series per time point, but which can be encoded as a single MF.

About transfer syntax, I am not so sure.

Reply:
I am not sure about transfer syntax either, but I think we could get into trouble if differently lossy compressed single-frame images should be combined into one multi-frame image.

Could we use Frame of Reference UID as distinguishing attribute?

Reply:
We are using FrameOfReferenceUID as a distinguishing attribute. I think should discuss this with @dclunie. Here is the [PixelMode](https://github.com/afshinmessiah/PixelMed/blob/master/pixelmed/dicom/FrameSet.java#L52) code which is the source of our code's logic.

Reply:
You mean we should discuss with David why he is not using transfer syntax as a distinguishing attribute?

Reply:
Yes I do.